python/src/etos_api/routers/v0/utilities.py

```diff
@@ -67,3 +73,36 @@ async def wait_for_artifact_created(etos_library, artifact_identity, artifact_id
         await asyncio.sleep(2)
     LOGGER.error("Artifact %r not found.", artifact_identifier)
     return None
+
+
+async def download_suite(test_suite_url: str) -> dict:
+    """Attempt to download suite.
+
+    :param test_suite_url: URL to test suite to download.
+    :return: Downloaded test suite as JSON.
+    """
+    try:
+        suite = requests.get(test_suite_url, timeout=60)
+        suite.raise_for_status()
+    except Exception as exception:  # pylint:disable=broad-except
+        raise AssertionError(f"Unable to download suite from {test_suite_url}") from exception
+    return suite.json()
+
+
+async def validate_suite(test_suite_url: str) -> None:
+    """Validate the ETOS test suite through the SuiteValidator.
+
+    :param test_suite_url: The URL to the test suite to validate.
+    """
+    span = trace.get_current_span()
+
+    try:
+        test_suite = await download_suite(test_suite_url)
+        await SuiteValidator().validate(test_suite)
+    except AssertionError as exception:
+        LOGGER.error("Test suite validation failed!")
+        LOGGER.error(exception)
+        span.add_event("Test suite validation failed")
```

Comment:
Record the exception.

Reply:
I did not change the code

Reply:
IF we do add recording of the exception I suggest removeing the `span.add_event` call, since it then becomes redundant.